projects/plugins/jetpack/tests/e2e/specs/post-connection/sync.test.js

```diff
@@ -0,0 +1,149 @@
+import { test, expect } from '../../fixtures/base-test.js';
+import { execWpCommand } from 'jetpack-e2e-commons/helpers/utils-helper.cjs';
+import {
+	enableSync,
+	disableSync,
+	enableDedicatedSync,
+	disableDedicatedSync,
+} from '../../helpers/sync-helper.js';
+import { BlockEditorPage } from 'jetpack-e2e-commons/pages/wp-admin/index.js';
+import { prerequisitesBuilder } from 'jetpack-e2e-commons/env/index.js';
+import playwrightConfig from '../../playwright.config.cjs';
+
+test.describe( 'Sync', () => {
+	const wpcomRestAPIBase = 'https://public-api.wordpress.com/rest/';
+	let blockEditor;
+	let wpcomBlogId;
+	let wpcomForcedPostsUrl;
+	let wpcomPostsResponse;
+	let wpcomPosts;
+
+	test.beforeAll( async ( { browser } ) => {
+		const page = await browser.newPage( playwrightConfig.use );
+		await prerequisitesBuilder( page ).withLoggedIn( true ).withConnection( true ).build();
+		await page.close();
+
+		const jetpackOptions = await execWpCommand( 'option get jetpack_options --format=json' );
+		wpcomBlogId = JSON.parse( jetpackOptions ).id;
+		wpcomForcedPostsUrl =
+			wpcomRestAPIBase + `v1/sites/${ wpcomBlogId }/posts?force=wpcom&search=Sync`;
+	} );
+
+	test.beforeEach( async ( { page } ) => {
+		await test.step( 'Visit block editor page', async () => {
+			blockEditor = await BlockEditorPage.visit( page );
+			await blockEditor.resolveWelcomeGuide( false );
+		} );
+	} );
+
+	test.afterEach( async () => {
+		await test.step( 'Reset Sync defaults', async () => {
+			await enableSync();
+			await disableDedicatedSync();
+		} );
+	} );
+
+	test( 'Sync', async ( { page } ) => {
```

Comment:
@brbrr @adimoldovan I noticed that the Sync related e2e tests were flaky and really wanted to avoid adding a timeout, so I took a different approach:

Instead of having 3 separate tests for each flow: Normal, Sync Disabled and Dedicated Sync I restructured the test to only one that will publish each post after setting the corresponding Sync flow and will call the API in the last step. I'm not crazy about this approach but I'm hoping that it will give Sync enough time to run and remove the flakiness. If you can think of a better approach please let me know.
(Also note that I haven't cleaned up the `sync.test.js` aka there's commented-out code and redundant code here but I'm planning to do so if we agree that the current approach is legit)

Reply:
To be honest, I'm not a fan of the new approach. While it buys a short amount of time for the first two sync actions, there is no guarantee that sync is done before the assertion.
I see no other way than checking that sync is done in a loop, with a few (milli)seconds of a wait between retries, until it's completed or until a reasonable timeout is reached. Maybe this "sync done" loop can check something simpler, like the number of posts, or anything else that could indicate that sync is (probably) done, and then assert the content of the post.